imagebuildah/build.go

```diff
@@ -431,3 +442,196 @@ func preprocessContainerfileContents(logger *logrus.Logger, containerfile string
+			if err != nil {
+				logrus.Debugf("parsing image reference %q: %v", candidate.Value.String(), err)
+				continue
+			}
+			src, err := ref.NewImageSource(ctx, systemContext)
+			if err != nil {
+				logrus.Debugf("preparing to read image manifest for %q: %v", baseImage, err)
+				continue
+			}
+			candidateBytes, candidateType, err := src.GetManifest(ctx, nil)
+			_ = src.Close()
+			if err != nil {
+				logrus.Debugf("reading image manifest for %q: %v", baseImage, err)
+				continue
+			}
+			if !manifest.MIMETypeIsMultiImage(candidateType) {
+				logrus.Debugf("base image %q is not a reference to a manifest list: %v", baseImage, err)
+				continue
+			}
+			if err := candidate.Record(); err != nil {
+				logrus.Debugf("error recording name %q for base image %q: %v", candidate.Value.String(), baseImage, err)
+				continue
+			}
+			baseImage = candidate.Value.String()
+			manifestBytes, manifestType = candidateBytes, candidateType
+			break
+		}
+		if len(manifestBytes) == 0 {
+			if len(resolved.PullCandidates) > 0 {
+				return nil, errors.Errorf("base image name %q didn't resolve to a manifest list", baseImage)
+			}
+			return nil, errors.Errorf("base image name %q didn't resolve to anything", baseImage)
+		}
+		if manifestType != manifest.DockerV2ListMediaType {
+			list, err := manifest.ListFromBlob(manifestBytes, manifestType)
+			if err != nil {
+				return nil, errors.Wrapf(err, "parsing manifest list from base image %q", baseImage)
+			}
+			list, err = list.ConvertToMIMEType(manifest.DockerV2ListMediaType)
+			if err != nil {
+				return nil, errors.Wrapf(err, "converting manifest list from base image %q to v2s2 list", baseImage)
+			}
+			manifestBytes, err = list.Serialize()
+			if err != nil {
+				return nil, errors.Wrapf(err, "encoding converted v2s2 manifest list for base image %q", baseImage)
+			}
+		}
+		s2list, err := manifest.Schema2ListFromManifest(manifestBytes)
+		if err != nil {
+			return nil, errors.Wrapf(err, "decoding manifest list for base image %q", baseImage)
+		}
+		if baseImageIndex == 0 {
+			// populate the list with the first image's platforms, with and without variant figured in
+			for _, instance := range s2list.Manifests {
+				platform := instance.Platform.OS + "/" + instance.Platform.Architecture
+				platforms[platform] = struct{}{}
+				if instance.Platform.Variant != "" {
+					platform = platform + "/" + instance.Platform.Variant
+				}
+				platforms[platform] = struct{}{}
+				logger.Debugf("image %q supports %q", baseImage, platform)
+			}
+		} else {
+			// prune the list of any platforms this base image doesn't support, with or without variant figured in
+			present := make(map[string]struct{})
+			for _, instance := range s2list.Manifests {
+				platform := instance.Platform.OS + "/" + instance.Platform.Architecture
+				present[platform] = struct{}{}
+				if instance.Platform.Variant != "" {
+					platform = platform + "/" + instance.Platform.Variant
```

Comment:
You want to build multiple arches with and without the variant?

Reply:
Should be fixed now.